src/lyra.ts

```diff
@@ -648,15 +649,22 @@ export function search<S extends PropertiesSchema>(
     for (let j = 0; j < tokensLength; j++) {
       const term = tokens[j];
       const documentIDs = getDocumentIDsFromSearch(lyra, { ...params, index, term, exact });
-      const termOccurrencies = lyraOccurrencies[term];
+      
+      // lyraOccurrencies[term] can be undefined, 0, string, or { [k: string]: number }
+      const termOccurrencies = typeof lyraOccurrencies[term] === "number"
+        ? (lyraOccurrencies[term] ?? 0)
+        : 0;
```

Comment:
- anomalies in `termOccurrencies`' value make `idf` become `NaN` or `Infinity`, with `tfIdf` resulting in a `NaN` value in turn
- when `termOccurrencies` is a string, object, or undefined -> `idf` becomes `NaN` -> `tfIdf` becomes `NaN`
- when `termOccurrencies` is 0 -> `idf` becomes `Infinity` -> `tfIdf` becomes `NaN`